adapter/internal/oasparser/model/common.go

```diff
@@ -197,3 +198,43 @@ func unmarshalSwaggerResources(path string, methods []*Operation, vendorExtensio
 		vendorExtensions: vendorExtensions,
 	}
 }
+
+// getRewriteRegexFromPathTemplate returns a regex with capture groups for given rewritePathTemplate
+func getRewriteRegexFromPathTemplate(pathTemplate, rewritePathTemplate string) (string, error) {
+	rewriteRegex := "/" + strings.TrimSuffix(strings.TrimPrefix(rewritePathTemplate, "/"), "/")
+	pathParamToIndexMap := getPathParamToIndexMap(pathTemplate)
+	r := regexp.MustCompile(`{uri.var.([^{}]+)}`)
+	matches := r.FindAllStringSubmatch(rewritePathTemplate, -1)
+	for _, match := range matches {
+		if len(match) > 1 {
```

Comment:
I think we can remove this check and the check in L235, since the length of slice is always two

Reply:
Removed with https://github.com/wso2/product-microgateway/pull/2786/commits/86e265b2f8e11e9df24abef4cccd005c1934c386